store/cachekv/store.go

```diff
@@ -171,6 +171,38 @@ func (store *Store) CacheWrapWithTrace(w io.Writer, tc types.TraceContext) types
 	return NewStore(tracekv.NewStore(store, w, tc))
 }
 
+// Copy creates a deep copy of the Store object
+func (store *Store) Copy() types.CacheKVStore {
```

Comment:
Can we call out that this will probably incur some overhead and should be used with caution?

We also need to test this